tests/cxx/typedef_chain_in_template-d4.h

```diff
@@ -0,0 +1,25 @@
+//===--- typedef_chain_in_template-d4.h - test input file for iwyu --------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// Container that tests typedef chains longer than in libc++.
+
+template<typename T>
+class ContainerLongTypedefChain {
+ public:
+  typedef T value_type;
+  typedef value_type& reference;
+  typedef reference& reference_reference;
```

Comment:
This is only valid in C++11, right? Or is it valid to form a universal/rvalue reference using nested typedefs? Apparently Clang is happy with it anyway...

I guess `-std=c++11` (or later) is enabled everywhere by default now, so it's probably OK.